docs/specs/delegation.md

```diff
@@ -0,0 +1,110 @@
+# Fabric Identities
+
+This document addresses future plans for identity and permissions for the Common Tools fabric, and the process to get there.
+
+## URL structure
+
+Via latest [PRD](https://docs.google.com/document/d/1KixOc7L5LZ8IdJtO_9pNHohPg_LXEF5hFgoAzvq9KLc/edit?tab=t.vx0btmvvbit9), the URL structure represents the three address component types:
+
+```
+DID = "did:key:{string}"
+DOMAIN = "{string.}*{string}.{string}"
+CHARM_ID = "of:${HASH}"
+NAMESPACE = DID | DOMAIN
+SPACE = DID | string;
+CHARM = CHARM_ID | string;
+ADDR = "/{@NAMESPACE/}?SPACE/CHARM?"
+```
+
+Each component may be a slug/name ("my-notes-123") or a DID key ("did:key:abc..").
+Namespaces are optional in the URL, and are prefixed with a "@" followed by either
+a DID key or DNS address. If namespace not provided, then an implicit, global provider-namespace is used.
+
+## Namespace
+
+Namespaces are a scope of spaces, and spaces contain charms.
+
+Namespaces can be referenced via DNS record, resolving to a DID key, or via directly as a DID key. Similarly, a provider-namespace is implied when no namespace given, which also resolves to an identity DID. 
+
+Namespaces manage many spaces, and are responsible from mapping space petnames to their identity.
+Each namespace has an "admin" space (possibly with the same identity as the namespace itself?), where records are stored.
+
+> [!NOTE]
+> There may be a way to view a namespace, like itemizing all contained spaces, but is out of scope here.
+
+Additionally, delegations for access to the namespace is stored in the admin space as well. See **CAPABILITIES** below for delegation.
+
+### Namespace Admin Space
+
+```
+DELEGATIONS = UCAN[];
+// Mapping of space name to space identity
+ITEMS = Map<string, DidKey>
+```
+
+### Provider Namespace
+
+The provider namespace is local to a provider service, and the default namespace used when none supplied. This functions like other user-owned namespaces, except owned by the provider.
+ 
+## Space
+
+Each space contains many charms and other data. A space is the root of permissions in the system, and capabilities are applied per space. Similar to namespaces, each space must maintain a mapping of charm names to identities, as well as permission for the space. Unlike namespaces, this data is stored in well-known Cells rather than a derivable space.
+
+### Space Admin Space
+
+```
+DELEGATIONS = UCAN[];
+// Mapping of charm name to charm identity
+ITEMS = Map<string, DidKey>
+```
+
+## Capabilities
+
+Access and permissions are handled via an authorization token like [UCAN](https://github.com/ucan-wg/spec) or another alternative. These tokens represent capabilities delegated to identities for a given subject.
+
+* `SPACE:CREATE`: A namespace-based capability indicating permission to create a new space. This is the only delegation stored in a namespace admin record. 
```

Comment:
+1, interestingly i think this could just correspond to the ability to create URIs whose hash is just derived from the space did, not some other URI that's already in the space. (assumes we restrict this, which we currently do, but sometimes consider since it guarantees unforgeable identifiers)